plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeLocalConcurrentWritesTest.java

```diff
@@ -247,6 +248,74 @@ private void testConcurrentInsertsSelectingFromTheSameTable(boolean partitioned)
         }
     }
 
+    @Test
+    public void testConcurrentInsertsSelectingFromTheSameVersionedTable()
+            throws Exception
+    {
+        testConcurrentInsertsSelectingFromTheSameVersionedTable(true);
+        testConcurrentInsertsSelectingFromTheSameVersionedTable(false);
+    }
+
+    private void testConcurrentInsertsSelectingFromTheSameVersionedTable(boolean partitioned)
+            throws Exception
+    {
+        int threads = 3;
+        CyclicBarrier barrier = new CyclicBarrier(threads);
+        ExecutorService executor = newFixedThreadPool(threads);
+        String tableName = "test_concurrent_inserts_select_from_same_versioned_table_" + randomNameSuffix();
+
+        assertUpdate("CREATE TABLE " + tableName + " (a, part) " + (partitioned ? " WITH (partitioned_by = ARRAY['part'])" : "") + "  AS VALUES (0, 10)", 1);
+
+        try {
+            // Considering T1, T2, T3 being the order of completion of the concurrent INSERT operations,
+            // if all the operations would eventually succeed, the entries inserted per thread would look like this:
```

Comment:
ALL the operations are expected to succeed - we're dealing with blind inserts here.
The only situation in which the operations wouldn't succeed would be when exhausting the retries - but we don't land in such a situation.